mewarpx/mewarpx/plotting.py

```diff
@@ -0,0 +1,633 @@
+        print("Saving plot image as", self.params["plot_name"] + ".jpg")
+        plt.savefig(self.params["plot_name"] + ".jpg")
+
+    def _gen_plot_contours(self):
+        """Generate the list of contours."""
+        if self.params['scale'] == 'linear':
+            return np.linspace(self.valmin, self.valmax,
+                               self.params['ncontours'])
+        # Make sure end points are captured, rather than missed due to floating
+        # point errors.
+        pmin = self.valmin - 1e-5*(self.valmax - self.valmin)
+        pthresh = np.abs(self.params["linthresh"])
+        pmax = self.valmax + 1e-5*(self.valmax - self.valmin)
+        # Try to have a proportional set of negative log contours, linear
+        # contours, and positive log contours. Linear is always 'linportion' of
+        # total # of contours here.
+        # Plot linear contours (any values in linear range)?
+        if pmin < pthresh and pmax > -pthresh:
+            lincontours = np.linspace(
+                max(pmin, -pthresh), min(pmax, pthresh),
+                int(round(self.params["ncontours"]*self.params["linportion"])))
+        else:
+            lincontours = np.array([])
+        nlogcontours = int(round(self.params["ncontours"] - len(lincontours)))
+        # Plot negative logarithmic contours?
+        if pmin < -pthresh:
+            nnegcontours = max(0, int(round(
+                ((np.log(-pmin) - np.log(pthresh))
+                 / (np.log(-pmin) + np.log(max(pthresh, abs(pmax)))
+                    - 2.*np.log(pthresh))
+                 * nlogcontours))))
+            neglogcontours = -1.0*np.exp(
+                np.linspace(np.log(pthresh), np.log(abs(pmin)), nnegcontours))
+        else:
+            nnegcontours = 0
+            neglogcontours = np.array([])
+        # Plot positive logarithmic contours?
+        if pmax > pthresh:
+            nposcontours = max(0, nlogcontours - nnegcontours)
+            poslogcontours = np.exp(np.linspace(np.log(pthresh), np.log(pmax),
+                                                nposcontours))
+        else:
+            poslogcontours = np.array([])
+        # Multiply by the appropriate factor, eliminate duplicates with set,
+        # then sort
+        contour_points = sorted(set(
+            np.concatenate([neglogcontours, lincontours, poslogcontours])))
+        return contour_points
+
+def get_vec(axis):
+    nxyz = (mwxrun.nx, mwxrun.ny, mwxrun.nz)
+    pos_lims = (mwxrun.xmin, mwxrun.xmax, mwxrun.ymin, mwxrun.ymax, mwxrun.zmin, mwxrun.zmax)
+
+    if axis == 'r':
+        raise ValueError("RZ plotting is not implemented yet.")
+        return self.get_rvec()
+    axis_dict = {0: 0, 1: 1, 2: 2, 'x': 0, 'y': 1, 'z': 2}
+    axis = axis_dict[axis]
+    npts = nxyz[axis]
+    xmin = pos_lims[2*axis]
+    xmax = pos_lims[2*axis + 1]
+    # There is one more point on the grid than cell number
+    return np.linspace(xmin, xmax, npts + 1)
+
+def recursive_update(d, u):
```

Comment:
Let's move this one to util too. It's incidental to plotting and not unlikely we'll use it elsewhere in the future too.